sharding/indexHashedNodesCoordinator.go

```diff
@@ -548,13 +475,30 @@ func (ihgs *indexHashedNodesCoordinator) ShardIdForEpoch(epoch uint32) (uint32,
 func (ihgs *indexHashedNodesCoordinator) GetConsensusWhitelistedNodes(
 	epoch uint32,
 ) (map[string]struct{}, error) {
-	publicKeysPrevEpoch := make(map[uint32][][]byte)
+
 	var err error
+	shardEligible := make(map[string]struct{})
+	publicKeysPrevEpoch := make(map[uint32][][]byte)
+	prevEpochConfigExists := false
 
 	if epoch > 0 {
 		publicKeysPrevEpoch, err = ihgs.GetAllValidatorsPublicKeys(epoch - 1)
-		if err != nil {
-			return nil, err
+		if err == nil {
+			prevEpochConfigExists = true
```

Comment:
all prevEpochConfigExists and prevEpochShardId calculation could be done inside this if.

Reply:
I do not want to have too much indentation inside that conditional

Reply:
It would have looked better. But can go as it is.